src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectAdapterProvider.cs

```diff
@@ -37,7 +32,7 @@ internal class VsProjectAdapterProvider : IVsProjectAdapterProvider
 
             _workspaceService = workspaceService;
             _threadingService = threadingService;
-            _vsSolution = new Lazy<IVsSolution>(() => serviceProvider.GetService<SVsSolution, IVsSolution>());
+            _vsSolution5 = new Lazy<IVsSolution5>(() => serviceProvider.GetService<SVsSolution, IVsSolution5>());
```

Comment:
Can you use `AsyncServiceProvider.GlobalProvider.GetServiceAsync()` to avoid possible deadlock risks in `serviceProvider.GetService()`, as stated in this [issue](https://github.com/NuGet/Home/issues/3784) ?

```suggestion
            _vsSolution5 = new AsyncLazy<IVsSolution5>(async () => {
                var svc = await AsyncServiceProvider.GlobalProvider.GetServiceAsync<SVsSolution>();
                return svc as IVsSolution5;
            }, ThreadHelper.JoinableTaskFactory);
```

Reply:
VsProjectAdapterProvider is now fully async.